metrics-exporter/src/bin/io_engine/client/replica_stat.rs

```diff
@@ -0,0 +1,79 @@
+use super::ticks_to_time;
+use serde::{Deserialize, Serialize};
+
+/// This stores IoStat information of a replica.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub(crate) struct ReplicaIoStat {
+    name: String,
+    entity_id: String,
+    bytes_read: u64,
+    num_read_ops: u64,
+    bytes_written: u64,
+    num_write_ops: u64,
+    read_latency_us: u64,
+    write_latency_us: u64,
+}
+
+impl ReplicaIoStat {
+    /// Get name of the replica.
+    pub(crate) fn name(&self) -> &String {
+        &self.name
+    }
+
+    /// Get used bytes read of the replica.
+    pub(crate) fn bytes_read(&self) -> u64 {
+        self.bytes_read
+    }
+
+    /// Get total number of read ops performed by the replica.
+    pub(crate) fn num_read_ops(&self) -> u64 {
+        self.num_read_ops
+    }
+
+    /// Get the total bytes written in bytes.
+    pub(crate) fn bytes_written(&self) -> u64 {
+        self.bytes_written
+    }
+
+    /// Get total number of write ops performed by the replica.
+    pub(crate) fn num_write_ops(&self) -> u64 {
+        self.num_write_ops
+    }
+
+    /// Get total read latency in usec on the replica.
+    pub(crate) fn read_latency_us(&self) -> u64 {
+        self.read_latency_us
+    }
+
+    /// Get total write latency in usec on the replica.
+    pub(crate) fn write_latency_us(&self) -> u64 {
+        self.write_latency_us
+    }
+
+    /// Get entity_id of the replica.
+    pub(crate) fn entity_id(&self) -> String {
+        self.entity_id.clone()
+    }
+}
+
+/// Array of NexusIoStat objects.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub(crate) struct ReplicaIoStats {
+    pub(crate) replica_stats: Vec<ReplicaIoStat>,
+}
+
+impl From<rpc::v1::stats::ReplicaIoStats> for ReplicaIoStat {
+    fn from(value: rpc::v1::stats::ReplicaIoStats) -> Self {
+        let stats = value.stats.unwrap();
+        Self {
+            name: stats.name,
+            entity_id: value.entity_id.unwrap(),
```

Comment:
hmm we can't do this, must use TryFrom...

Reply:
Okay